sphinxcontrib/jsonschema.py

```diff
@@ -38,6 +40,21 @@ def custom_jsonref_jsonloader(uri, **kwargs):
     return {}
 
 
+def urnjsonloader(load_path, uri, **kwargs):
```

Comment:
Have refactored this a bit to be a closure rather than a partial (so hopefully a bit more explicit about what is going on), but since the callable is being passed into jsonref which we don't control, and needs to know about the directory where schema files are located (which isn't a parameter we can pass), it is going to need to be a closure/partial/class with callable instances, or something similar to make this work.